faculty/clients/environment.py

```diff
@@ -42,16 +186,77 @@ class EnvironmentSchema(BaseSchema):
     author_id = fields.UUID(data_key="authorId", required=True)
     created_at = fields.DateTime(data_key="createdAt", required=True)
     updated_at = fields.DateTime(data_key="updatedAt", required=True)
+    specification = fields.Nested(SpecificationSchema(), required=True)
+
+    @post_load
+    def make_environment(self, data):
+        return Environment(**data)
+
+
+class EnvironmentUpdateSchema(BaseSchema):
+    name = fields.String(required=True)
+    description = fields.String(required=True)
+    specification = fields.Nested(SpecificationSchema(), required=True)
+
+    @post_load
+    def make_environment_update(self, data):
+        return EnvironmentUpdate(**data)
+
+
+class EnvironmentCreationResponseSchema(BaseSchema):
+    id = fields.UUID(data_key="environmentId", required=True)
+    project_id = fields.UUID(data_key="projectId", required=True)
+    name = fields.String(required=True)
+    description = fields.String(required=True)
+    author_id = fields.UUID(data_key="authorId", required=True)
+    specification = fields.Nested(SpecificationSchema(), required=True)
 
     @post_load
     def make_environment(self, data):
+        data["created_at"] = None
+        data["updated_at"] = None
```

Comment:
Returning an environment that's almost fully formed but not quite is a bit awkward.

I think there's an argument for only returning the ID.

Reply:
I completely agree - I assume that it's easier for me to just return the environment ID than for you to modify the environment creation endpoint, so I'll do that. (Though I do think these two fields should be returned by the environment creation endpoint if possible, for consistency.)

Reply:
@acroz @zblz what do you think?